src/accelerate/utils/modeling.py

```diff
@@ -1486,17 +1493,12 @@ def load_state_dict(checkpoint_file, device_map=None):
                 progress_bar = None
             for device in devices:
                 target_device = device
-
-                if is_xpu_available():
-                    current_safetensors_version = packaging.version.parse(importlib.metadata.version("safetensors"))
-
-                    if compare_versions(current_safetensors_version, "<", "0.4.2"):
-                        raise ModuleNotFoundError(
-                            f"You need at least safetensors 0.4.2 for Intel GPU, while you have {current_safetensors_version}"
+                if isinstance(device, int) and is_xpu_available():
+                    if compare_versions("safetensors", "<", "0.4.2"):
+                        raise ImportError(
+                            "Safetensors version must be >= 0.4.2 for Intel GPU. Please update safetensors."
```

Comment:
I suppose device int judgement and xpu judgement should be parallel, tell me the reason you put it w/ and.

Reply:
why parallel? as a reference: https://github.com/huggingface/accelerate/blob/1f7a79b428749f45187ec69485f2c966fe21926e/src/accelerate/big_modeling.py#L482  but I put the integer check first, because if it is not an integer, no meaning to check xpu device. But this might not be the conversion.